python/pyarrow/types.pxi

```diff
@@ -1009,6 +1471,18 @@ cdef class PyExtensionType(ExtensionType):
     ----------
     storage_type : DataType
         The storage type for which the extension is built.
+
+    Examples
+    --------
+    Define a UuidType extension type subclassing PyExtensionType:
+
+    >>> import pyarrow as pa
+    >>> class UuidType(pa.PyExtensionType):
+    ...     def __init__(self):
+    ...         pa.PyExtensionType.__init__(self, pa.binary(16))
+    ...     def __reduce__(self):
+    ...         return UuidType, ()
+    ...
```

Comment:
Not sure why, but doctest fails if I proceed with the `PyExtensionType` example and add `>>> uuid_type = UuidType()`.

The failure is:

```python
_______________________________________________________________________________________________________________________ [doctest] pyarrow.lib.PyExtensionType _______________________________________________________________________________________________________________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> uuid_type = UuidType()
UNEXPECTED EXCEPTION: PicklingError("Can't pickle <class 'pyarrow.lib.UuidType'>: attribute lookup UuidType on pyarrow.lib failed")
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.16/Frameworks/Python.framework/Versions/3.9/lib/python3.9/doctest.py", line 1334, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest pyarrow.lib.PyExtensionType[2]>", line 1, in <module>
  File "<doctest pyarrow.lib.PyExtensionType[1]>", line 3, in __init__
  File "pyarrow/types.pxi", line 1498, in pyarrow.lib.PyExtensionType.__init__
  File "pyarrow/types.pxi", line 1399, in pyarrow.lib.ExtensionType.__init__
  File "pyarrow/types.pxi", line 1405, in pyarrow.lib.ExtensionType.init
  File "pyarrow/types.pxi", line 1505, in pyarrow.lib.PyExtensionType.__arrow_ext_serialize__
_pickle.PicklingError: Can't pickle <class 'pyarrow.lib.UuidType'>: attribute lookup UuidType on pyarrow.lib failed
/Users/alenkafrim/repos/arrow/python/pyarrow/lib.cpython-39-darwin.so:None: UnexpectedException
```

Will look into it.

Reply:
Ah, seems doctest and pickle do not get along well :D
(related: https://stackoverflow.com/questions/73394835/how-to-test-pickle-ability-in-a-doctest)

Will add a skip here and what is missing to have the example complete.